src/models/dashboard-items.js

```diff
@@ -94,27 +99,32 @@ const dataMapper = (dataList, filterList) =>
 
 export function itemsMetadataConstructor(dashboardItemsMetadata) {
     const { dataElements, indicators, antigenCategory } = dashboardItemsMetadata;
-    const {
-        tablesDataCodes: { vaccinesTable, qsIndicatorsTable },
-        chartsDataCodes: { indicatorChart, utilizationRateChart },
-    } = dashboardItemsConfig;
+    const { tablesDataCodes, chartsDataCodes } = dashboardItemsConfig;
 
-    const vaccineTableDE = dataMapper(dataElements, vaccinesTable);
-    const qsTableDE = dataMapper(dataElements, qsIndicatorsTable);
-    const indicatorChartInd = dataMapper(indicators, indicatorChart);
-    const utilizationRateChartInd = dataMapper(indicators, utilizationRateChart);
+    const chartKeys = _.keys(chartsDataCodes);
+    const tableKeys = _.keys(tablesDataCodes);
+
+    const { vaccinesTable, qsIndicatorsTable } = tableKeys.reduce(
+        (acc, tb) => ({ ...acc, [tb]: dataMapper(dataElements, tablesDataCodes[tb]) }),
+        {}
+    );
```

Comment:
That's a common pattern, there is an abstraction in lodash for this: _.fromPairs. Something like this (untested):

```
 const { vaccinesTable, qsIndicatorsTable } = _(tablesDataCodes)
        .map((codes, key) => [key, dataMapper(dataElements, codes)])
        .fromPairs()
        .value();
```

Note how _.map can map objects (value/key, not key/value as args!), while there is no obj.map in JS.

Reply:
great!